src/core/nRF52832/serial.ads

```diff
@@ -0,0 +1,168 @@
+   procedure Initialize;
+   procedure Print (Str : String);
+   procedure Send;
+
+private
+
+   function String_Address (S : String) return System.Address;
+
+   type CONFIG_Parity is (Excluded, Included) with
+     Size => 3;
+
+   for CONFIG_Parity use (Excluded => 0,
+                          Included => 7);
+
+   type Reg_CONFIG is record
+      HWFC   : Boolean;
+      PARITY : CONFIG_Parity;
+   end record with
+     Size => 32;
+
+   for Reg_CONFIG use record
+      HWFC at 0 range 0 .. 0;
+      PARITY at 0 range 1 .. 3;
+   end record;
+
+   type PSEL_TXD_Connect is (Connected, Disconnected) with
+     Size => 1;
+
+   for PSEL_TXD_Connect use (Connected    => 0,
+                             Disconnected => 1);
+   --  UARTE
+   type Reg_PSELTXD is record
+      PIN : GPIO.Pin;
+   end record with
+     Size => 32;
+
+   --  UARTE
+   for Reg_PSELTXD use record
+      PIN at 0 range 0 .. 4;
+   end record;
+
+   type Reg_PSEL_TXD is record
+      PIN     : GPIO.Pin;
+      CONNECT : PSEL_TXD_Connect;
+   end record with
+     Size => 32;
+
+   for Reg_PSEL_TXD use record
+      PIN at 0 range 0 .. 4;
+      CONNECT at 0 range 31 .. 31;
+   end record;
+
+   type ENABLE_Enable is (Disabled, Enabled_UART, Enabled_UARTE) with
+     Size => 4;
+
+   for ENABLE_Enable use (Disabled => 0,
+                          Enabled_UART  => 4,
+                          Enabled_UARTE => 8);
+
+   type Reg_ENABLE is record
+      ENABLE : ENABLE_Enable;
+   end record;
+
+   for Reg_ENABLE use record
+      ENABLE at 0 range 0 .. 3;
+   end record;
+
+   type Reg_TXD_PTR is record
+      PTR : System.Address;
+   end record with
+     Size => 32;
+
+   type Count is range 0 .. 255 with
+     Size => 8;
+
+   type Reg_TXD_MAXCNT is record
+      MAXCNT : Count;
+   end record with
+     Size => 32;
+
+   for Reg_TXD_MAXCNT use record
+      MAXCNT at 0 range 0 .. 7;
+   end record;
+
+   type Reg_TXD_AMOUNT is record
+      AMOUNT : Count;
+   end record with
+     Size => 32;
+
+   for Reg_TXD_AMOUNT use record
+      AMOUNT at 0 range 0 .. 7;
+   end record;
+
+   type EVENT_Event is (Clear, Set) with
+     Size => 1;
+
+   for EVENT_Event use (Clear => 0,
+                        Set   => 1);
+
+   type Reg_EVENT is record
+      EVENT : EVENT_Event;
+   end record with
+     Size => 32;
+
+   for Reg_EVENT use record
+      EVENT at 0 range 0 .. 0;
+   end record;
+
+   type TASK_Trigger is (Trigger) with
+     Size => 1;
+
+   for TASK_Trigger use (Trigger => 1);
+
+   type Reg_TASK is record
+      TSK : TASK_Trigger;
+   end record with
+     Size => 32;
+
+   for Reg_TASK use record
+      TSK at 0 range 0 .. 0;
+   end record;
+
+   type BAUDRATE_Rate is (Baud1200, Baud2400, Baud4800, Baud9600,
+                          Baud14400, Baud19200, Baud28800, Baud38400,
+                          Baud57600, Baud76800, Baud115200, Baud230400,
+                          Baud250000, Baud460800, Baud921600, Baud1M) with
+     Size => 32;
```

Comment:
Defining this as an enum may be problematic when reading a `BAUDRATE_Rate` value from a register. Should this register have an invalid value (i.e. one not present in the representation clause) we may get an exception. Do we ever read a value of this type?

Reply:
This value should never be read.

Reply:
Then this should be safe.